lib/events/pgevents/common/utils.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pgcommon
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/utils/retryutils"
+)
+
+// TODO(espadolini): delete in favor of lib/backend/pgbk/common
+
+// ConnectPostgres will open a single connection to the "postgres" database in
+// the database cluster specified in poolConfig.
+func ConnectPostgres(ctx context.Context, poolConfig *pgxpool.Config) (*pgx.Conn, error) {
+	connConfig := poolConfig.ConnConfig.Copy()
+	connConfig.Database = "postgres"
+
+	if poolConfig.BeforeConnect != nil {
+		if err := poolConfig.BeforeConnect(ctx, connConfig); err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+
+	conn, err := pgx.ConnectConfig(ctx, connConfig)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if poolConfig.AfterConnect != nil {
+		if err := poolConfig.AfterConnect(ctx, conn); err != nil {
+			conn.Close(ctx)
+			return nil, trace.Wrap(err)
+		}
+	}
+
+	return conn, nil
+}
+
+// TryEnsureDatabase will connect to the "postgres" database and attempt to
+// create the database named in the pool's configuration.
+func TryEnsureDatabase(ctx context.Context, poolConfig *pgxpool.Config, log logrus.FieldLogger) {
+	pgConn, err := ConnectPostgres(ctx, poolConfig)
+	if err != nil {
+		log.WithError(err).Warn("Failed to connect to the \"postgres\" database.")
+		return
+	}
+
+	// the database name is not a string but an identifier, so we can't use query parameters for it
+	createDB := fmt.Sprintf("CREATE DATABASE \"%v\" TEMPLATE template0 ENCODING UTF8 LC_COLLATE 'C' LC_CTYPE 'C'", poolConfig.ConnConfig.Database)
```

Comment:
What if I change the name in the configuration after it's already been created?

Reply:
There's no way to change the configuration without restarting the audit log (or Teleport as a whole, really).